pinot-core/src/main/java/org/apache/pinot/core/query/reduce/ExecutionStatsAggregator.java

```diff
@@ -89,7 +90,7 @@ public void aggregate(ServerRoutingInstance routingInstance, DataTable dataTable
   public synchronized void aggregate(@Nullable ServerRoutingInstance routingInstance, Map<String, String> metadata,
       Map<Integer, String> exceptions) {
     // Reduce on trace info.
-    if (_enableTrace) {
+    if (_enableTrace && metadata.containsKey(DataTable.MetadataKey.TRACE_INFO.getName())) {
```

Comment:
why do we need both checks?

Reply:
NullPointerException without the second check. Reason being the map doesn't contain traceInfo even when tracing i enabled for V2Engine.

Reply:
i see. we should clean this up later. it is not easy to track the code when we have multiple way to determine the same config. 
either we 
1. dont set _enableTrace as member variable and always deduce from metadata
2. make _enableTrace correct without needing to look at metadata